hrms/hr/doctype/leave_policy_assignment/leave_policy_assignment.py

```diff
@@ -260,9 +259,7 @@ def calculate_pro_rated_leaves(
 
 	leaves *= actual_period / complete_period
 
-	if is_earned_leave:
-		return flt(leaves, precision)
-	return rounded(leaves)
+	return flt(leaves, precision)
```

Comment:
pro-rated calculations are always rounded off in most cases because this calculation is done as per the number of days and not months.

![image](https://github.com/frappe/hrms/assets/24353136/4a4f8e9a-3454-40c5-a08a-93f197c1c70e)

Check the failing test case for example.

Annual Allocation = 12
If an employee joins at the beginning of the 3rd month, **they should still get 9 whole leaves for the remaining 9 months**. But with this change, they will get **8.992**, hence the last leave cannot be used completely. 

Changing this behavior now would mean a breaking change. IMO, this should go behind a feature flag in Leave Type:
**Round pro rated leaves for first-year-allocation**
This should be hidden if leave type = earned. Should be enabled by default to continue ongoing behavior.

Reply:
Closing this. A separate feature PR can be opened for the same.